manager/scheduler/constraint.go

```diff
@@ -44,29 +44,49 @@ func (f *ConstraintFilter) Check(n *NodeInfo) bool {
 			if !constraint.Match(n.Description.Hostname) {
 				return false
 			}
-		default:
-			// default is node label in form like 'node.labels.key==value'
-			// if it is not well formed, always fails it
-			if !strings.HasPrefix(constraint.Key, "node.labels.") {
+		case "node.name":
+			if !constraint.Match(n.Spec.Annotations.Name) {
 				return false
 			}
-			// if the node doesn't have any label,
-			// it's equivalent to match an empty value.
-			// that is, 'node.labels.key!=value' should pass and
-			// 'node.labels.key==value' should fail
-			if n.Spec.Annotations.Labels == nil {
-				if !constraint.Match("") {
+		case "node.role":
+			if !constraint.Match(n.Spec.Role.String()) {
+				return false
+			}
+		default:
+			// default is label constraint in form like 'node.labels.key==value'
+			// or 'engine.labels.key!=value'
+			if strings.HasPrefix(constraint.Key, "node.labels.") {
```

Comment:
Would prefer if this prefix was defined as a constant, since it's also used below. It's stylistic, but I'm always afraid of typos making repeated strings inconsistent.